src/suggestions.js

```diff
@@ -0,0 +1,83 @@
+import {computeAccessibleName} from 'dom-accessibility-api'
+import {getRoles} from './role-helpers'
+import {getDefaultNormalizer} from './matches'
+
+const normalize = getDefaultNormalizer()
+
+function getLabelTextFor(element) {
+  let label
+
+  const allLabels = Array.from(document.querySelectorAll('label'))
+
+  label = allLabels.find(lbl => lbl.control === element)
+
+  if (!label) {
+    const ariaLabelledBy = element.getAttribute('aria-labelledby')
+
+    if (ariaLabelledBy) {
+      // we're using this notation because with the # selector we would have to escape special characters e.g. user.name
+      // see https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector#Escaping_special_characters
+      label = document.querySelector(`[id=${ariaLabelledBy}]`)
+    }
+  }
+
```

Comment:
I *think* we could do this instead:

```suggestion
  const label = element.labels.find(label => Boolean(normalize(label.textContent)))
```

Reply:
oh neat, is that like the inverse of label.control?  I was wondering if something like that exists.  i'll try this one after the kids go to bed. :)

Reply:
@kentcdodds was able to use that for the first bit, but for ariaLabelledBy it didn't work so I had to keep that part.